src/System.Private.ServiceModel/tools/PRService/sync.ashx

```diff
@@ -0,0 +1,227 @@
+            };
+
+            RunGitCommands(gitCommands, executionResult);
+        }
+        catch (Exception ex)
+        {
+            executionResult.AppendFormat("Exception caught while attempting to start '{0}'", _gitExecutablePath);
+            executionResult.Append(ex.ToString());
+            return false;
+        }
+
+        return true;
+    }
+
+    private bool SyncToPr(string pr, StringBuilder executionResult)
+    {
+        string[] gitCommands = new string[]
+        {
+            "clean -fdx",
+            "checkout master",
+            "fetch --all --prune",
+            "merge --ff-only origin/master",
+            string.Format("fetch origin pull/{0}/head:pr/{0}", pr),
+            string.Format("checkout pr/{0}", pr)
+        };
+
+        return RunGitCommands(gitCommands, executionResult);
+    }
+
+    private bool SyncToSha(string sha, StringBuilder executionResult)
+    {
+        string[] gitCommands = new string[]
+        {
+            "clean -fdx",
+            "checkout master",
+            "fetch --all --prune",
+            "merge --ff-only origin/master",
+            string.Format("checkout {0}", sha)
+        };
+
+        return RunGitCommands(gitCommands, executionResult);
+    }
+
+    private void CleanupMergedBranches(StringBuilder executionResult)
+    {
+        RunGitCommands(new string[] { "checkout master" }, executionResult);
+
+        StringBuilder branches = new StringBuilder();
+        RunGitCommands(new string[] { "branch" }, branches);
+
+        var branchesList =
+            branches.ToString().Split(new char[] { '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries);
+
+        List<string> commands = new List<string>();
+
+        for (int i = 0; i < branchesList.Length; i++)
+        {
+            // Don't try to delete "master" branch, since we're checked out to it
+            if (!branchesList[i].EndsWith("master"))
+            {
+                commands.Add(string.Format("branch -D {0}", branchesList[i]));
+            }
+        }
+
+        RunGitCommands(commands.ToArray(), executionResult);
+    }
+
+    private bool RunGitCommands(string[] gitCommands, StringBuilder executionResult)
+    {
+        ProcessStartInfo psi = new ProcessStartInfo()
+        {
+            CreateNoWindow = true,
+            FileName = _gitExecutablePath,
+            RedirectStandardError = true,
+            RedirectStandardOutput = true,
+            UseShellExecute = false,
+            WorkingDirectory = _gitRepoPath
+        };
+
+        bool success = false;
+
+        foreach (string gitCommand in gitCommands)
+        {
+            psi.Arguments = gitCommand;
+
+            Process p = Process.Start(psi);
+            if (!p.WaitForExit(_gitExecutionTimeoutMilliseconds))
+            {
+                executionResult.Append(string.Format("Git executable '{0}' took more than '{1}' ms to execute", _gitExecutablePath, _gitSanityExecutionTimeoutMilliseconds));
```

Comment:
I guess it's supposed to be _gitExecutionTimeoutMilliseconds instead of _gitSanityExecutionTimeoutMilliseconds? Looks like _gitSanityExecutionTimeoutMilliseconds will be dead code afterwards.